### ISSUE TITLE
OoT Settings: Ganon's Boss Key

### ISSUE THREAD

**Issue by Revvenn** [OoTMM/core#69]
In OOTR the Ganon's Boss Key has its own drop-down options for its requirements or settings, treated like a goal like the Rainbow Bridge. Ganon's Boss Key in OOTR, for settings that aren't merely basic shuffles, has a special function of being given to the player anywhere rather than locked to a specific check for requirements like Dungeon Rewards or Medallions or Skull Tokens etc. I suggest looking at this PR for how it was done: https://github.com/TestRunnerSRL/OoT-Randomizer/pull/1367

Now to the point, the settings:

-Remove (Keysy). Self-explanatory, it removes the boss key from the pool and the door is unlocked. Most players use this setting in OOTR.
-Vanilla Location
-Own Dungeon
-Regional (this is specific to OOTR in that it uses the hint regions for where Ganon's Boss Key can be, a more restricted form of keysanity, probably something that can't be implemented right now but much later on when we have hint regions set up so for now ignore this one I guess)
-Overworld Only (shuffles Ganon's Boss Key to anywhere in the overworld only)
-Any Dungeon (shuffles Ganon's Boss Key to be inside Any Dungeon)
-Anywhere (Keysanity)
-Light Arrow Cutscene (meaning tied to the Spirit and Shadow Medallions)
-Dungeon Rewards
-Stones
-Medallions
-Hearts
-Gold Skull Tokens

Basically, Dungeon Rewards, Stones, Medallions, Hearts, and Tokens all have configurable sliders like the Rainbow Bridge and upon completing those goals you will receive Ganon's Boss Key no matter where you are in the game. Think of it like you're playing multiworld and someone sends you an item. By allowing you to set the GBK to a separate goal you can then open Ganon's Castle and have it more involved earlier in the seed which could be fun in OOTxMM while still locking away Ganondorf and having to do dungeons and what not.

**Comment by Nax** [comment]
A basic version of this was implemented, will re-open/work on other options once keysanity gets added.
